### ISSUE TITLE
feature request for default or global instrumentation 

### ISSUE THREAD

**Issue by blackluck** [open-telemetry/opentelemetry-operator#1804]
Hello!

It would be a nice option to have a default instrumentation cr for the k8s cluster, for example setting a root namespace (like where operator is installed by default) and by default operator use that instrumentation config if there isn't any config in the namespace where you want to autoinstrument you app.
For a bigger cluster with lots of namespaces, especially when mostly you need the same instrumentation cr it would be much easier to configure it only once, modify that one only if needed etc.

Thanks

**Comment by jaronoff97** [comment]
This is a good idea, and one we should investigate for the next iteration of resources @open-telemetry/operator-approvers

**Comment by TylerHelmuth** [comment]
Agreed, if we decided that the operator should have a fallback instrumentation. But I don't think it should.

If I added `instrumentation.opentelemetry.io/inject-dotnet: "true"` to my app, how do I know which Instrumentation resource did the injection?  Is there one in my namespace and I got it or did I fallback to a default value? There might be something in the operator logs or k8s events, but as an app team I may not know to go look for those.

I much prefer the auto-instrumentation having explicit rules. At the moment `instrumentation.opentelemetry.io/inject-dotnet: "true"` can only have 1 interpretation for where the instrumentation came from, but with a fallback it becomes ambiguous.  I think it gets even weirder if a user does `instrumentation.opentelemetry.io/inject-dotnet: "my-instrumentation"` and there is not Instrumentation named `my-instrumentation`.

Since the idea of a cluster-wide Instrumentation object is already supported via the namespaced-selection (`instrumentation.opentelemetry.io/inject-dotnet: "monitoring/cluster-wide-instrumentation"`), I'd vote we don't implement this feature. In this scenario you only have to configure 1 instrumentation object, which is a request of the Issue, and any app can use it by referencing it directly in the required opt-in annotation.
